libyul/AsmParser.cpp

```diff
@@ -76,11 +65,38 @@ std::shared_ptr<DebugData const> Parser::createDebugData() const
 	switch (m_useSourceLocationFrom)
 	{
 		case UseSourceLocationFrom::Scanner:
-			return DebugData::create(ParserBase::currentLocation());
+			return DebugData::create(ParserBase::currentLocation(), ParserBase::currentLocation());
 		case UseSourceLocationFrom::LocationOverride:
-			return DebugData::create(m_locationOverride);
+			return DebugData::create(m_locationOverride, m_locationOverride);
```

Comment:
> 2. location override: This is used for snippets of yul code generated from solidity code in the old codegen. It probably makes sense for both to point at the solidity source location.

I think that `m_locationOverride` for both is the most consistent. If I use current location for `nativeLocation` instead, `Parser` will be reporting errors at `m_locationOverride` but `AsmAnalyzer` (which, I'm assuming gets locations from `nativeLocation` in the AST) will be be reporting them at the current location.

> We could also make the ir location point at the scanner location, but we probably don't have the actual source anymore at the point where we generate a potential parser error.

Errors are processed and reported from within `CompilerContext::appendInlineAssembly()` which still has the original snippet. To check that I introduced an error to one of the generated sources and there were no failing assertions and messages seemed to be pointing at the correct location in the snippet - even when `nativeLocation` was set to the current location rather than the override.

But there's still the problem with inconsistent location in errors I mentioned above. We can change that but we should then also change error messages to consistently use one kind of locations.

Reply:
It probably doesn't matter too much, because these will all lead to internal compiler errors anyway.